PR: https://github.com/odoo/odoo/pull/165935

From: 65652c73ba5331f77911186543062dfddf50f077
From: Manushi Shah
Date: 2024-05-23 11:01:05

Structural Changes: 8
Total Changes: 8

[IMP] product, *: improve product form UX

This commit do following UI changes:
- Update string of sales and purchase tax also set
purchase tax from sales tax.
- Hide purchase tax for no purchase_ok field.
- Change strings of some fields for better UX.
- Add POS page in product.
- Re~sequence state button in product.
- Hide tax and other field for combo products.
- Change string goods to consumable.

task-3526304

Part-of: odoo/odoo#165935
Co-authored-by: Kartik Chavda <kcv@odoo.com>

================================= pseudo patch: =================================

--- a/addons/account/models/product.py
+++ b/addons/account/models/product.py
@@ -29,12 +29,16 @@ class ProductCategory(models.Model):
 class ProductTemplate(models.Model):
     _inherit = "product.template"
 
-    taxes_id = fields.Many2many('account.tax', 'product_taxes_rel', 'prod_id', 'tax_id', help="Default taxes used when selling the product.", string='Customer Taxes',
+    taxes_id = fields.Many2many('account.tax', 'product_taxes_rel', 'prod_id', 'tax_id',
+        string="Sales Taxes",
+        help="Default taxes used when selling the product",
         domain=[('type_tax_use', '=', 'sale')],
         default=lambda self: self.env.companies.account_sale_tax_id or self.env.companies.root_id.sudo().account_sale_tax_id,
     )
     tax_string = fields.Char(compute='_compute_tax_string')
-    supplier_taxes_id = fields.Many2many('account.tax', 'product_supplier_taxes_rel', 'prod_id', 'tax_id', string='Vendor Taxes', help='Default taxes used when buying the product.',
+    supplier_taxes_id = fields.Many2many('account.tax', 'product_supplier_taxes_rel', 'prod_id', 'tax_id',
+        string="Purchase Taxes",
+        help="Default taxes used when buying the product",
         domain=[('type_tax_use', '=', 'purchase')],
         default=lambda self: self.env.companies.account_purchase_tax_id or self.env.companies.root_id.sudo().account_purchase_tax_id,
     )
